src/org/labkey/test/components/ui/grids/EditableGrid.java

```diff
@@ -529,6 +530,20 @@ else if (value instanceof LocalTime localTime)
         return gridCell;
     }
 
+    public void setEntityData(List<Map<String, Object> >data, List<FieldDefinition> fields)
+    {
+        for (int i = 0; i < data.size(); i++)
+        {
+            Map<String, Object> rowData = data.get(i);
+            for (FieldDefinition field : fields) {
+                String key = field.getLabel() != null ? field.getLabel() : field.getName();
```

Comment:
It's beyond the scope of this PR, but I'm wondering if it would be worthwhile to make a change to FieldDefinition to have a method like `getIdentifier` which would have this conditional.

May not be worth it, but just an idea.

Reply:
I had a similar thought. (And, sorry, I saw this after I pushed the Merge button.) Certainly worth revision in a later PR.